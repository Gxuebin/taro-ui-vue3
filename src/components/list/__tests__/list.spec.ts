import { mount } from '@vue/test-utils'
import AtList from '../index'
import AtListItem from '../item'

describe('List', () => {
  it('should render the complete List', () => {
    const wrapper = mount({
      data() {
        return {
          iconInfo: {
            size: 25,
            color: '#78A4FA',
            value: 'calendar',
          },
        }
      },
      components: {
        AtList,
        AtListItem,
      },
      template: `
    <AtList>
      <AtListItem title='标题文字' />
      <AtListItem title='标题文字' arrow='right' />
      <AtListItem title='标题文字' note='描述信息' />
      <AtListItem title='禁用状态' disabled extraText='详细信息' />
      <AtListItem title='标题文字' note='描述信息' arrow='right' />
      <AtListItem title='标题文字' extraText='详细信息' arrow='right' />
      <AtListItem
        arrow='right'
        note='描述信息'
        title='标题文字标题文字标题文字标题文字标题文字'
        extraText='详细信息详细信息详细信息详细信息'
      />
      <AtListItem
        title='标题文字'
        note='描述信息'
        extraText='详细信息'
        arrow='right'
        thumb='http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png'
      />
      <AtListItem
        arrow='right'
        note='描述信息'
        :iconInfo="iconInfo"
        title='标题文字'
        extraText='详细信息'
        thumb='http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png'
      />
      <AtListItem title='标题文字' isSwitch />
      <AtListItem title='标题文字' isSwitch disabled />
      <AtListItem title='标题文字' switchIsCheck isSwitch />
      <AtListItem title='标题文字' switchIsCheck isSwitch disabled />
    </AtList>
      `,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the completed List -- no border', () => {
    const wrapper = mount({
      components: {
        AtList,
        AtListItem,
      },
      template: `
      <AtList :hasBorder="false">
        <AtListItem title='标题文字' :hasBorder="false" />
        <AtListItem title='标题文字' :hasBorder="false" />
      </AtList>
      `,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('List Behavior ', () => {
  it('should trigger onClick event on ListItem', () => {
    const onClick = jest.fn()
    const wrapper = mount({
      components: {
        AtList,
        AtListItem,
      },
      methods: {
        onClick: onClick,
      },
      template: `
      <AtList>
        <AtListItem title='标题文字' @click="onClick" />
      </AtList>
      `,
    })
    wrapper.find('.at-list .at-list__item').trigger('tap')
    expect(onClick).toBeCalled()
  })

  it('should trigger onSwitchChange event on ListItem', async () => {

    const wrapper = mount({
      data: () => ({
        checked: false
      }),
      components: {
        AtList,
        AtListItem,
      },
      methods: {
        onSwitchChange: jest.fn().mockImplementation((e) => {
          wrapper.vm.$data.checked = e.detail
        })
      },
      template: `
        <AtList>
          <AtListItem
            title='标题文字'
            is-switch
            :switch-is-check="checked"
            @switch-change="onSwitchChange"
          />
        </AtList>
      `,
    })
    expect(wrapper.vm.$data.checked).toBeFalsy()
    wrapper.find('.at-list .at-list__item .item-extra__switch switch').trigger('change', { detail: true })
    expect(wrapper.vm.$data.checked).toBeTruthy()
  })

  it('ListItem switch should be checked', () => {
    const wrapper = mount({
      components: {
        AtList,
        AtListItem,
      },
      template: `
        <AtList>
          <AtListItem
            title='标题文字'
            is-switch
            switch-is-check
          />
        </AtList>
      `,
    })
    const switchEl = wrapper.find('.at-list .at-list__item .item-extra__switch switch')
    expect(switchEl.attributes('checked')).toBe("true")
  })

  it('ListItem switch should be unchecked', () => {
    const wrapper = mount({
      components: {
        AtList,
        AtListItem,
      },
      template: `
        <AtList>
          <AtListItem
            title='标题文字'
            is-switch
          />
        </AtList>
      `,
    })
    const switchEl = wrapper.find('.at-list .at-list__item .item-extra__switch switch')
    expect(switchEl.attributes('checked')).toBe("false")
  })
})
