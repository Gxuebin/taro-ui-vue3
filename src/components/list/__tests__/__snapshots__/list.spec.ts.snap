// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the complete List 1`] = `
<view
  class="at-list"
>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <!---->
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <!---->
        <view
          class="item-extra__icon"
        >
          <view
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </view>
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--multiple at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          <view
            class="item-content__info-note"
          >
            描述信息
          </view>
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <!---->
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--disabled at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            禁用状态
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        <view
          class="item-extra__info"
        >
          详细信息
        </view>
        
        <!---->
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--multiple at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          <view
            class="item-content__info-note"
          >
            描述信息
          </view>
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <!---->
        <view
          class="item-extra__icon"
        >
          <view
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </view>
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        <view
          class="item-extra__info"
        >
          详细信息
        </view>
        
        <!---->
        <view
          class="item-extra__icon"
        >
          <view
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </view>
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--multiple at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字标题文字标题文字标题文字标题文字
          </view>
          <view
            class="item-content__info-note"
          >
            描述信息
          </view>
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        <view
          class="item-extra__info"
        >
          详细信息详细信息详细信息详细信息
        </view>
        
        <!---->
        <view
          class="item-extra__icon"
        >
          <view
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </view>
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--thumb at-list__item--multiple at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      <view
        class="at-list__item-thumb item-thumb"
      >
        <image
          class="item-thumb__info"
          mode="scaleToFill"
          src="http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png"
        />
      </view>
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          <view
            class="item-content__info-note"
          >
            描述信息
          </view>
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        <view
          class="item-extra__info"
        >
          详细信息
        </view>
        
        <!---->
        <view
          class="item-extra__icon"
        >
          <view
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </view>
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--thumb at-list__item--multiple at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      <view
        class="at-list__item-thumb item-thumb"
      >
        <image
          class="item-thumb__info"
          mode="scaleToFill"
          src="http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png"
        />
      </view>
      <view
        class="at-list__item-icon item-icon"
      >
        <view
          class="at-icon at-icon-calendar"
          style="color: rgb(120, 164, 250); font-size: 25px;"
        />
      </view>
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          <view
            class="item-content__info-note"
          >
            描述信息
          </view>
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        <view
          class="item-extra__info"
        >
          详细信息
        </view>
        
        <!---->
        <view
          class="item-extra__icon"
        >
          <view
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </view>
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <view
          class="item-extra__switch"
        >
          <switch
            checked="false"
            color="#6190E8"
            disabled="false"
          />
        </view>
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--disabled at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <view
          class="item-extra__switch"
        >
          <switch
            checked="false"
            color="#6190E8"
            disabled="true"
          />
        </view>
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <view
          class="item-extra__switch"
        >
          <switch
            checked="true"
            color="#6190E8"
            disabled="false"
          />
        </view>
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--disabled at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <view
          class="item-extra__switch"
        >
          <switch
            checked="true"
            color="#6190E8"
            disabled="true"
          />
        </view>
        
      </view>
    </view>
  </view>
</view>
`;

exports[`List should render the completed List -- no border 1`] = `
<view
  class="at-list at-list--no-border"
>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <!---->
        
      </view>
    </view>
  </view>
  <view
    class="at-list__item at-list__item--no-border"
  >
    <view
      class="at-list__item-container"
    >
      
      
      <view
        class="at-list__item-content item-content"
      >
        <view
          class="item-content__info"
        >
          <view
            class="item-content__info-title"
          >
            标题文字
          </view>
          
        </view>
      </view>
      <view
        class="at-list__item-extra item-extra"
      >
        
        
        <!---->
        
      </view>
    </view>
  </view>
</view>
`;
